### PR TITLE
Covered targets issue

### PR DESCRIPTION
**Pull request by ZhangMan1126**

### CODE REVIEW

File: core/src/main/kotlin/org/evomaster/core/search/service/Archive.kt [hunk 1]
```diff
@@ -385,7 +385,13 @@ class Archive<T> where T : Individual {
             }
 
             val limit = apc.getArchiveTargetLimit()
-            if (current.size < limit) {
+
+            /**
+             individual can be added only if the target k is not covered.
+             If a target is covered and a 'better'(e.g., shorter) individual appears,
+             it would be handled as replacement, see line 404
+             */
```
Comment (arcuri82) [inline core/src/main/kotlin/org/evomaster/core/search/service/Archive.kt]: do not refer to a line by number, as that ll likely change at the next modification of this file :)

File: core/src/main/kotlin/org/evomaster/core/search/service/Archive.kt [hunk 2]
```diff
@@ -385,7 +385,13 @@ class Archive<T> where T : Individual {
             }
 
             val limit = apc.getArchiveTargetLimit()
-            if (current.size < limit) {
+
+            /**
+             individual can be added only if the target k is not covered.
```
Comment (arcuri82) [inline core/src/main/kotlin/org/evomaster/core/search/service/Archive.kt]: inside a function, no need to use JavaDoc notation `/**`, just use `/*`

File: core/src/main/kotlin/org/evomaster/core/search/service/Archive.kt (hunk 1 above)
Comment (ZhangMan1126) [reply core/src/main/kotlin/org/evomaster/core/search/service/Archive.kt]: modified :)

File: core/src/main/kotlin/org/evomaster/core/search/service/Archive.kt (hunk 2 above)
Comment (ZhangMan1126) [reply core/src/main/kotlin/org/evomaster/core/search/service/Archive.kt]: modified

**Pull request merged**